Connect smart lock to end points ES-147

Patrick-250 opened the pull request:
1. After the user logs in,  the backend returns the tenantId  in the response.
2, This tenantID is dispatched to redux store
3.settenantID action updates the tenant slice with the ID
4.useselector hook retrieve the tenant Id and use in in LockUnlockCard
5, the tenantId is also used in apicall to fetch the current state for specific tenant(when logged in). 

                                                             ****How to test****
1.log in and interact with smartlock card.. the lock/unlock should update in locks table in supabase..(for this to work, the backend functionality must be merged first or at least u have to run its backend branch  locally


                                                               ***Logged In***
![image](https://github.com/user-attachments/assets/96e2098f-a0b9-4d58-87e6-b408ef4d8054)
![image](https://github.com/user-attachments/assets/f5593034-ca8a-46bc-b272-fd1f4edcc0d0)

![image](https://github.com/user-attachments/assets/5c689405-749e-4246-b1e7-bc43238cf2e4)
![image](https://github.com/user-attachments/assets/331e2753-62f4-4d4d-88a2-b855796dcba9)

Patrick-250 committed:
feat:Connectsmart-locks to backend

Patrick-250 committed:
add import.meta.env.VITE_API_BASE_URL

Patrick-250 committed:
Merge branch 'dev' into Connect-Smart-Lock-to-end-points

Patrick-250 committed:
fix:remove non logged in user to interact with smart lock, this would have been handled anyways since non tenants won't access dashboard

Patrick-250 committed:
Merge branch 'Connect-Smart-Lock-to-end-points' of https://github.com/EliteSpace-DSD/EliteSpace-Frontend into Connect-Smart-Lock-to-end-points

jasminepvo reviewed (changes requested):
I am logged in and still seeing this 
<img width="1212" alt="Screenshot 2025-03-24 at 10 35 36 PM" src="https://github.com/user-attachments/assets/1d340990-e6b4-4b63-b015-96f33f1aed7a" />

Patrick-250 commented:
> I am logged in and still seeing this <img alt="Screenshot 2025-03-24 at 10 35 36 PM" width="1212" src="https://private-user-images.githubusercontent.com/99847030/426345705-1d340990-e6b4-4b63-b015-96f33f1aed7a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI4NzYwMzEsIm5iZiI6MTc0Mjg3NTczMSwicGF0aCI6Ii85OTg0NzAzMC80MjYzNDU3MDUtMWQzNDA5OTAtZTZiNC00YjYzLWIwMTUtOTZmMzNmMWFlZDdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI1VDA0MDg1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ1NWUzNWM4OTdhMjUyMjM1NzQxNDcxOWM2Mzk3MGNjMGIyMTJlNzViZDhmZGNiMTIwZDY0YjNmY2EyNDcxMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qVjaTda-ADBFgjjCtr4j49LyG822HHcSGoGIDTrh3kU">

this is because you are running a different branch on backend branch. i have  a PR for backend that needs to be approved first. or you can test locally by running backend branch at "Smart-Locks-Backend"

Patrick-250 committed:
Merge branch 'dev' into Connect-Smart-Lock-to-end-points

Patrick-250 committed:
remove duplicate import

Patrick-250 commented:
> I was on that branch in the backend. When I tested backend with the existing tenant id provided, I was able to see the correct response shown in your instructions
> 
> However when I tested frontend with backend running, I got that error. I'll screen record a video for you to see

I just retested this and works on my end! I would love to see whats happening on your end.

Patrick-250 committed:
Merge branch 'dev' into Connect-Smart-Lock-to-end-points

jasminepvo commented:
Tested and working locally, I will also approve https://github.com/EliteSpace-DSD/EliteSpace-BackEnd/pull/34
<img width="1207" alt="Screenshot 2025-03-25 at 1 03 40 PM" src="https://github.com/user-attachments/assets/21edcd38-a6b9-4906-a514-25202afedc93" />
<img width="1207" alt="Screenshot 2025-03-25 at 1 03 30 PM" src="https://github.com/user-attachments/assets/1b434336-927d-4bbc-a31d-85c8551b4c09" />

jasminepvo reviewed (approved):
Great work Patrick!

Patrick-250 committed:
feat: Persist tenantId in localStorage for session persistence

- Updated tenantSlice to load tenantId from localStorage during initialization.
- Modified setTenantId action to save tenantId to localStorage for persistence across browser refreshes.
- Added clearTenantId action to remove tenantId from both Redux state and localStorage during logout.

Patrick-250 committed:
Merge branch 'dev' into Connect-Smart-Lock-to-end-points
